src/server/debug.rs

```diff
@@ -732,110 +731,82 @@ impl Debugger {
 
     pub fn get_region_properties(&self, region_id: u64) -> Result<Vec<(String, String)>> {
         let region_state = self.get_region_state(region_id)?;
-        let region = region_state.get_region();
-        let db = &self.engines.kv;
+        dump_properties(&self.engines.kv, region_state.get_region())
+    }
 
-        let mut num_entries = 0;
-        let mut mvcc_properties = MvccProperties::new();
-        let start = keys::enc_start_key(&region);
-        let end = keys::enc_end_key(&region);
-        let collection = box_try!(db.c().get_range_properties_cf(CF_WRITE, &start, &end));
-        for (_, v) in collection.iter() {
-            num_entries += v.num_entries();
-            let mvcc = box_try!(MvccProperties::decode(&v.user_collected_properties()));
-            mvcc_properties.add(&mvcc);
-        }
-
-        let middle_key = match box_try!(get_region_approximate_middle(db.c(), &region)) {
-            Some(data_key) => {
-                let mut key = keys::origin_key(&data_key);
-                box_try!(bytes::decode_bytes(&mut key, false))
-            }
-            None => Vec::new(),
-        };
+    pub fn get_range_properties(&self, start: &[u8], end: &[u8]) -> Result<Vec<(String, String)>> {
+        let mut region = Region::default();
```

Comment:
I think we can call `get_range_properties` in `get_region_properties`, and let `get_range_properties` dump properties based on range instead of region. Construct a region here is weird.

As for the `middle_key_by_approximate_size` is calculated based on region, you can just handle it in `get_region_properties` individually and append to result from `get_range_properties`